test/test_resume.cpp

```diff
@@ -1690,3 +1690,42 @@ TORRENT_TEST(resume_data_have_pieces)
 	TEST_EQUAL(rs->params.unfinished_pieces.size(), 1);
 }
 
+// See https://github.com/arvidn/libtorrent/issues/5174
+TORRENT_TEST(removed)
+{
+	// we're _likely_ to trigger the condition, but not guaranteed. loop
+	// until we do.
+	while (true) {
+		lt::session ses(settings());
+		std::shared_ptr<torrent_info> ti = generate_torrent();
+		add_torrent_params p;
+		p.ti = ti;
+		p.save_path = ".";
+		torrent_handle h = ses.add_torrent(p);
+		// this is asynchronous
+		ses.remove_torrent(h);
+		try {
+			h.save_resume_data();
+		} catch (std::exception const& e) {
+			std::printf("failed to trigger condition, retrying\n");
+			continue;
+		}
+		// we triggered the condition.
+		while (true)
```

Comment:
I don't think this infinite loop is a good idea either. I think it's better to try for some amount of time, and failing if the condition doesn't happen within some reasonable timeout. (in cases where we expect to always see the condition that is)